packages/perseus/src/util/graphie-utils.ts

```diff
@@ -0,0 +1,253 @@
+            const {data, localized} = entry;
+            onDataLoaded(data, localized);
+        } else {
+            entry.dataCallbacks.push(onDataLoaded);
+        }
+    } else {
+        const cacheData = {
+            loaded: false as const,
+            dataCallbacks: [onDataLoaded],
+            localized: shouldUseLocalizedData(),
+        };
+
+        labelDataCache[hash] = cacheData;
+
+        const retrieveData = async (
+            url: string,
+            errorCallback: (x?: any, status?: any, error?: any) => void,
+        ) => {
+            const response = await fetch(url);
+
+            if (!response.ok) {
+                errorCallback();
+                return;
+            }
+
+            const jsonp = await response.text();
+
+            const data = parseDataFromJSONP(hash, jsonp, (error) => {
+                Log.error(
+                    "Failed to parse JSONP for svg-image",
+                    Errors.Service,
+                    {
+                        cause: error,
+                        loggedMetadata: {
+                            dataUrl: Util.getDataUrl(url),
+                            jsonp,
+                        },
+                    },
+                );
+            });
+
+            // If the data is null, then we failed to parse the JSONP
+            if (!data) {
+                return;
+            }
+
+            const newCacheEntry = (labelDataCache[hash] = {
+                ...labelDataCache[hash],
+                loaded: true as const,
+                data,
+            });
+
+            newCacheEntry.dataCallbacks.forEach((callback) => {
+                callback(newCacheEntry.data, cacheData.localized);
+            });
+        };
+
+        if (shouldUseLocalizedData()) {
+            retrieveData(getLocalizedDataUrl(url), (x, status, error) => {
+                cacheData.localized = false;
+
+                // If there is isn't any localized data, fall back to
+                // the original, unlocalized data
+                retrieveData(Util.getDataUrl(url), (x, status, error) => {
+                    Log.error(
+                        "Data load failed for svg-image",
+                        Errors.Service,
+                        {
+                            cause: error,
+                            loggedMetadata: {
+                                dataUrl: Util.getDataUrl(url),
+                                status,
+                            },
+                        },
+                    );
+                });
+            });
+        } else {
+            retrieveData(Util.getDataUrl(url), (x, status, error) => {
+                Log.error("Data load failed for svg-image", Errors.Service, {
+                    cause: error,
+                    loggedMetadata: {
+                        dataUrl: Util.getDataUrl(url),
+                        status,
+                    },
+                });
+            });
+        }
```

Comment:
I understand that this is simply copy-pasted from the `svg-image.tsx` file, but I think a bit of simplification might be helpful here. Perhaps the `shouldUseLocalizedData()` conditional block could be written as follows:
```
const dataErrorHandler = (x, status, error) => {
    Log.error("Data load failed for svg-image", Errors.Service, {
        cause: error,
        loggedMetadata: {
            dataUrl: Util.getDataUrl(url),
            status,
        },
});

if (shouldUseLocalizedData()) {
    retrieveData(getLocalizedDataUrl(url), (x, status, error) => {
        cacheData.localized = false;

        // If there is isn't any localized data, fall back to
        // the original, unlocalized data
        retrieveData(Util.getDataUrl(url), dataErrorHandler)
} else {
    retrieveData(Util.getDataUrl(url), dataErrorHandler)
}
```